Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx

```diff
@@ -146,17 +146,21 @@ const SubmissionsProvider = ({ children }: ISubmissionsProviderProps) => {
     useEffect(
         () => {
             const { allowedSubmissionTypes } = currentProblem || {};
-            const submissionType = first(allowedSubmissionTypes);
+            const submissionType = allowedSubmissionTypes?.find((st) => st.id === selectedSubmissionType?.id);
 
-            if (submissionType) {
-                const { id } = submissionType;
+            if (isNil(submissionType)) {
+                const defaultSubmissionType = first(allowedSubmissionTypes);
 
-                selectSubmissionTypeById(id);
-            } else {
-                selectSubmissionTypeById(null);
+                return isNil(defaultSubmissionType)
+                    ? selectSubmissionTypeById(null)
+                    : selectSubmissionTypeById(defaultSubmissionType.id);
             }
+
+            const { id } = submissionType;
+
+            return selectSubmissionTypeById(id);
```

Comment:
The whole `useEffect` can be done as follows:

```
const { allowedSubmissionTypes } = currentProblem || {};
const submissionType = allowedSubmissionTypes?.find((st) => st.id === selectedSubmissionType?.id);

const { id } = submissionType || first(allowedSubmissionTypes) || { id: null };
selectSubmissionTypeById(id);
```

Reply:
Great, refactored!